src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/PackageManagerUIRefreshEvent.cs

```diff
@@ -3,36 +3,99 @@
 
 using System;
 using NuGet.Common;
+using NuGet.VisualStudio.Internal.Contracts;
 
 namespace NuGet.PackageManagement.Telemetry
 {
     public sealed class PackageManagerUIRefreshEvent : TelemetryEvent
     {
         private const string EventName = "PMUIRefresh";
 
-        public PackageManagerUIRefreshEvent(
+        private PackageManagerUIRefreshEvent(
             Guid parentId,
             bool isSolutionLevel,
             RefreshOperationSource refreshSource,
             RefreshOperationStatus refreshStatus,
             string tab,
             bool isUIFiltering,
             TimeSpan timeSinceLastRefresh,
-            double? duration) : base(EventName)
+            double? duration,
+            string projectId = null,
+            NuGetProjectKind? projectKind = null) : base(EventName)
         {
             base["ParentId"] = parentId.ToString();
             base["IsSolutionLevel"] = isSolutionLevel;
             base["RefreshSource"] = refreshSource;
             base["RefreshStatus"] = refreshStatus;
-            base["Tab"] = tab;
+            base["Tab"] = tab ?? throw new ArgumentNullException(nameof(tab));
             base["IsUIFiltering"] = isUIFiltering;
             base["TimeSinceLastRefresh"] = timeSinceLastRefresh.TotalMilliseconds;
 
+            if (!isSolutionLevel)
+            {
+                if (projectId != null)
+                {
+                    base["ProjectId"] = projectId;
+                }
+                if (projectKind.HasValue)
+                {
+                    base["ProjectKind"] = projectKind;
+                }
+            }
+
             if (duration.HasValue)
             {
                 base["Duration"] = duration;
             }
         }
+
+        public static PackageManagerUIRefreshEvent ForProject(
+            Guid parentId,
+            RefreshOperationSource refreshSource,
+            RefreshOperationStatus refreshStatus,
+            string tab,
+            bool isUIFiltering,
+            TimeSpan timeSinceLastRefresh,
+            double? duration,
+            string projectId,
+            NuGetProjectKind projectKind)
+        {
+            var evt = new PackageManagerUIRefreshEvent(
+                parentId,
+                isSolutionLevel: false,
+                refreshSource,
+                refreshStatus,
+                tab,
+                isUIFiltering,
+                timeSinceLastRefresh,
+                duration,
+                projectId,
```

Comment:
Does it go through PID anonymization step?

Reply:
Investigating with data team.

Reply:
It should not, it's just guids.
We already collect this data in other places, we shouldn't overthink it :)

Reply:
It's EUPI if checking this [link](https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/226/Telemetry-Catalog-How-should-I-classify-my-telemetry-point-(GDPR-Classification)).
And our [OneNote](https://microsoft.sharepoint.com/teams/NuGet/_layouts/OneNote.aspx?id=%2Fteams%2FNuGet%2FTeam%2FNuGet&wd=target%28Engineering.one%7C219AABAA-18B3-46F4-83E8-D3188D83DE22%2FGDPR%20Inventory%7CCCC0B159-9948-43E9-A618-4ADB05C5AB52%2F%29) mentioned how to process EUPI.